ext/oj/dump.c

```diff
@@ -856,6 +856,8 @@ void oj_dump_cstr(const char *str, size_t cnt, bool is_sym, bool escape1, Out ou
                 str = dump_unicode(str, end, out, orig);
                 break;
             case '6':  // control characters
+                if ((uint8_t)*str == 0 && out->opts->skip_null_byte == Yes) break;
+
```

Comment:
> For efficiency the check for a value of zero should be inside the check for less than 0x80 to avoid two conditional checks when the character is over 0x80 as in UTF8 characters.

Excellent catch. Haven't thought of that.

> Also please stick with the existing style and put the break on a separate line and use {}.

I know I read this in `README` 🤦 :
- Code is now formatted with the clang-format tool with the configuration file in the root of the repo.

> C is a bit different than say Ruby, Go, or others in that at least on older compiles *str = 0 is accepted as an assignment and ends up with a value of 0 which is false. Many old school C developer reverse the statement to 0 == *str. You will see that throughout the Oj code. Please use that ordering for the two conditions being checked.

Didn't know about this. Thank you very much for detailed explanation!

Will fix ASAP!

Reply:
Done: https://github.com/ohler55/oj/pull/883/commits/74c650d7e8515347308670fd877f97507dab80cb